worldedit-core/src/main/java/com/sk89q/worldedit/extension/factory/parser/pattern/ClipboardPatternParser.java

```diff
@@ -28,31 +28,53 @@
 import com.sk89q.worldedit.extent.clipboard.Clipboard;
 import com.sk89q.worldedit.function.pattern.ClipboardPattern;
 import com.sk89q.worldedit.function.pattern.Pattern;
-import com.sk89q.worldedit.internal.registry.SimpleInputParser;
+import com.sk89q.worldedit.internal.registry.InputParser;
+import com.sk89q.worldedit.math.BlockVector3;
 import com.sk89q.worldedit.session.ClipboardHolder;
 
 import java.util.List;
 
-public class ClipboardPatternParser extends SimpleInputParser<Pattern> {
+public class ClipboardPatternParser extends InputParser<Pattern> {
 
     public ClipboardPatternParser(WorldEdit worldEdit) {
         super(worldEdit);
     }
 
     @Override
-    public List<String> getMatchedAliases() {
+    public List<String> getSuggestions() {
         return Lists.newArrayList("#clipboard", "#copy");
     }
 
     @Override
-    public Pattern parseFromSimpleInput(String input, ParserContext context) throws InputParseException {
+    public Pattern parseFromInput(String input, ParserContext context) throws InputParseException {
+        if (!input.startsWith("#clipboard") && !input.startsWith("#copy")) {
+            return null;
+        }
         LocalSession session = context.requireSession();
 
+        int offsetPart;
+        BlockVector3 offset = BlockVector3.ZERO;
+        if ((offsetPart = input.indexOf('@')) >= 0) {
```

Comment:
I'd just use split here instead of indexOf+substring.
Makes for more readable code.